core/shared/src/main/scala/fs2/Hotswap.scala

```diff
@@ -108,13 +108,13 @@ object Hotswap {
       new Hotswap[F, R] {
         override def swap(next: Resource[F, R]): F[R] =
           // workaround for https://github.com/typelevel/cats-effect/issues/579
-          (next <* ().pure[Resource[F, *]]).allocated
-            .continual { r => // this whole block is inside continual and cannot be canceled
+          Concurrent[F].continual((next <* ().pure[Resource[F, *]]).allocated) {
```

Comment:
Dotty gets confused by the call to `.allocated` - this issue shows up again in the test suite. One solution is just ascribing a type. Without this change:

```scala
[error] -- [E008] Not Found Error: /Users/mpilquist/Development/oss/fs2/core/shared/src/main/scala/fs2/Hotswap.scala:111:54
[error] 111 |          (next <* ().pure[Resource[F, *]]).allocated.continual {
[error]     |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]     |value continual is not a member of F[(R, F[Unit])], but could be made available as an extension method.
[error]     |
[error]     |One of the following imports might make progress towards fixing the problem:
[error]     |
[error]     |  import cats.effect.Concurrent.ops.toAllConcurrentOps
[error]     |  import cats.effect.ConcurrentEffect.ops.toAllConcurrentEffectOps
[error]     |
[error]     |
[error]     |
[error]     |where:    F is a type in method create with bounds <: [_$3] =>> Any
[error] one error found
```